doc/source/grouping.rst

```diff
@@ -0,0 +1,145 @@
+        segment.analogsignals.append(signal)
+
+
+Now a more complex example: a 1x4 silicon probe, with a neuron on channels 0,1,2 and another neuron on channels 1,2,3.
+We create a :class:`ChannelIndex` for each neuron to hold the :class:`Unit` object associated with this spike sorting group.
+Each :class:`ChannelIndex` also contains the list of channels on which that neuron spiked.
+
+::
+
+    import numpy as np
+    from quantities import ms, mV, kHz
+    from neo import Block, Segment, ChannelIndex, Unit, SpikeTrain, AnalogSignal
+
+    block = Block(name="probe data")
+    segment = Segment()
+    segment.block = block
+    block.segments.append(segment)
+
+    # create 4-channel AnalogSignal with dummy data
+    signal = AnalogSignal(np.random.rand(1000, 4) * mV,
+                          sampling_rate=10 * kHz)
+    # create spike trains with dummy data
+    # we will pretend the spikes have been extracted from the dummy signal
+    spiketrains = [
+        SpikeTrain(np.arange(5, 100) * ms, t_stop=100 * ms),
+        SpikeTrain(np.arange(7, 100) * ms, t_stop=100 * ms)
+    ]
+    segment.analogsignals.append(signal)
+    segment.spiketrains.extend(spiketrains)
+    # assign each spiketrain to a neuron (Unit)
+    units = []
+    for i, spiketrain in enumerate(spiketrains):
+        unit = Unit(name=f"Neuron #{i + 1}")
+        unit.spiketrains.append(spiketrain)
+        units.append(unit)
+
+    # create a ChannelIndex for each unit, to show which channels the spikes come from
+    chx0 = ChannelIndex(name="Channel Group 1", index=[0, 1, 2])
+    chx0.units.append(units[0])
+    chx0.analogsignals.append(signal)
+    units[0].channel_index = chx0
+    chx1 = ChannelIndex(name="Channel Group 2", index=[1, 2, 3])
+    chx1.units.append(units[1])
+    chx1.analogsignals.append(signal)
+    units[1].channel_index = chx1
+
+    block.channel_indexes.extend((chx0, chx1))
+
+
+Using :class:`View` and :class`Group`::
+
+    import numpy as np
+    from quantities import ms, mV, kHz
+    from neo import Block, Segment, View, Group, SpikeTrain, AnalogSignal
+
+    block = Block(name="probe data")
+    segment = Segment()
+    segment.block = block
+    block.segments.append(segment)
+
+    # create 4-channel AnalogSignal with dummy data
+    signal = AnalogSignal(np.random.rand(1000, 4) * mV,
+                          sampling_rate=10 * kHz)
+    # create spike trains with dummy data
+    # we will pretend the spikes have been extracted from the dummy signal
+    spiketrains = [
+        SpikeTrain(np.arange(5, 100) * ms, t_stop=100 * ms),
+        SpikeTrain(np.arange(7, 100) * ms, t_stop=100 * ms)
+    ]
+    segment.analogsignals.append(signal)
+    segment.spiketrains.extend(spiketrains)
+    # assign each spiketrain to a neuron (now using Group)
+    units = []
+    for i, spiketrain in enumerate(spiketrains):
+        unit = Group(spiketrain, name=f"Neuron #{i + 1}")
+        units.append(unit)
+
```

Comment:
I think this example would more clear with sevral segment.
a unit is more easy to understand across segment. No ?

Reply:
This example is just a direct translation of the previous one, it's probably not the best example for the new features.

We should write new examples when we rewrite the documentation.